### PR TITLE
Refactor formatting logic

### PR DESCRIPTION
**Pull request by CraigFe**
An assorted collection of improvements/refactorings aimed at extracting more logic from the `Core` module. Mostly consists of splitting out pretty-printers for the output values and keeping them separate from the `Core.t` type, which is currently very overloaded.

### CODE REVIEW

File: src/alcotest/pprint.mli [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+type result = {
+  success : int;
+  failures : int;
+  time : float;
+  errors : string list;
+}
+
+val pp_rresult_error : run_result Fmt.t
+
+val pp_event :
```
Comment (samoht) [inline src/alcotest/pprint.mli]: `PPrint.pp_event` has a lot of `p`s :-) What about naming this module `Pp` and dropping the `pp_` prefix to have `Pp.event` ?

Comment (CraigFe) [reply src/alcotest/pprint.mli]: Excellent idea; done.

**Comment by CraigFe** [comment]
This is now based on https://github.com/mirage/alcotest/pull/196 for CI convenience.

**Comment by samoht** [comment]
Can you rebase now that #196 has been merged?

**Comment by CraigFe** [comment]
Done :slightly_smiling_face:

**Pull request merged**